### DISCUSSION TITLE
Unable to load lsp backend for js-mode or js2-mode

### DISCUSSION THREAD

**Discussion by bobdobbs** [syl20bnr/spacemacs#15326] [Q&A]
I've got the javascript layer running and have installed lsp-mode

In order to set up lsp-mode, I've been following the documention provided here:
https://develop.spacemacs.org/layers/+lang/javascript/README.html#backends

Following the documentation, I've added these lines to my spacemacs config.
(I've also tried removing them my spacemacs file and moving them to ~/.emacs.d/init.el. This gets the same results)

```
    ((js-mode (javascript-backend . lsp)))
   ((js2-mode (javascript-backend . lsp)))
```

Starting spacemacs with --debug-init yields this message:

Debugger entered--Lisp error: (invalid-function (js2-mode (javascript-backend . lsp)))

The full message is [as follows](https://gist.github.com/bobdobbs/8d7a845c55007a070375b7e843b334af)

**Comment by lebensterben** [comment]
Please post your .spacemacs

**Comment by bobdobbs** [comment]
> Please post your .spacemacs

https://gist.github.com/bobdobbs/a5cdf858293a6f72d65e235d3567cd9f

**Comment by lebensterben** [comment]
First,
Where in the documentation do you find it telling you to have
```
    ((js-mode (javascript-backend . lsp)))
   ((js2-mode (javascript-backend . lsp)))
```

They are not valid Lisp expressions.

---

Second, all user configuration should not pass the last line of `.spacemacs` as can seen [here](https://github.com/syl20bnr/spacemacs/blob/d366150139c2c3fa8b743f502efccd542e7a6f3a/core/templates/.spacemacs.template#L555-L556).

**Reply by bobdobbs** [reply]
>> Where in the documentation do you find it telling you to have

This is under "2.5 Choosing a backend"

```
;;; Directory Local Variables
;;; For more information see (info "(emacs) Directory Variables")

((js2-mode (javascript-backend . lsp)))
```

**Reply by lebensterben** [reply]
In this very [section](https://develop.spacemacs.org/layers/+lang/javascript/README.html#choosing-a-backend)

> lsp backend will be automatically chosen if the layer lsp is used and you did not specify any value for javascript-backend

So you don't have to set anything if you want to use `lsp` as backend.

---

As for the code you injected, 

First there's no mentioning of `js-mode`. You should not invent it by yourself.
Second, just above the code example it says:

> Backend can be chosen on a per project basis using directory local variables (files named .dir-locals.el at the root of a project)

So it clearly says they're used as directory local variable, which are set in `.dir-locals.el` at the root of a project, ***NOT*** in your `.spacemacs`.
